adr/minimal-sdk.md

```diff
@@ -0,0 +1,53 @@
+6. **New, Minimal Red Hat Container/Service:** Provide an ultra-lightweight container that exposes a simple HTTP API for agent interaction and recommend users graduate to more powerful frameworks for complex needs.
+7. **New, Minimal Red Hat SDK:** Provide an ultra-lightweight SDK designed for basic use cases and tutorials, while officially recommending that customers adopt comprehensive third-party frameworks for advanced applications.
+
+## Decision
+
+We will adopt **Option 7**. We will create a **new, minimal, lightweight Red Hat Agent SDK** and concurrently establish a strategy to **officially recommend and ensure compatibility with established third-party frameworks**.
+
+The new Red Hat Agent SDK will be written in **Python**. The specific list of third-party frameworks we will recommend and ensure compatibility with will be addressed in one or more future ADRs.
+
+The key characteristics of the Red Hat Agent SDK will be:
+
+* **Enablement, Not Competition:** Its primary purpose is to serve as a clear, simple "golden path" for new users, tutorials, documentation, and basic proof-of-concept applications.
+* **Minimalist Scope:** It will focus exclusively on implementing the core agent loop (state management, API calls, tool dispatching) as a reference. It is not intended to have complex features like chains, memory modules, or vector store integrations.  The details of the scope will be defined by the capabilities we want to introduce in our enablement and/or any automatic generation of agentic code that we do.
+* **Reference Implementation:** It will act as the reference for how to correctly interact with the Llama Stack Responses API in an agentic fashion, providing a foundation that other tools can build upon or refer to.  It may be useful as a reference for developers who want to build their own Agent abstraction for a single application.
+
+This two-pronged approach (a minimal "first-step" SDK and a clear "next-step" recommendation of powerful external tools) provides the best balance of speed, user choice, and focused engineering effort.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Rapid Time-to-Market:** A minimal SDK is a fast way to deliver a functional agent construct to our users, allowing us to provide a complete end-to-end story quickly.
+* **Empowers User Choice:** This strategy explicitly embraces the rich AI ecosystem. Customers with existing expertise in established frameworks can use their preferred tools, while new users have a simple, non-intimidating entry point.
+* **Low Engineering Overhead:** By strictly limiting the scope of our own SDK, we avoid the massive maintenance burden of creating and supporting a full-featured framework. Engineering efforts are focused and contained.
```

Comment:
What would that scope be?

Reply:
This is covered in the Decision section which states:
> **Minimalist Scope:** It will focus exclusively on implementing the core agent loop (state management, API calls, tool dispatching) as a reference. It is not intended to have complex features like chains, memory modules, or vector store integrations.  The details of the scope will be defined by the capabilities we want to introduce in our enablement and/or any automatic generation of agentic code that we do.